bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -5,33 +5,61 @@ import java.io.IOException
 /**
  * Capture and serialize the state of all threads at the time of an exception.
  */
-internal class ThreadState @JvmOverloads constructor
-    (
-    config: ImmutableConfig,
+internal class ThreadState @JvmOverloads constructor(
     exc: Throwable?,
+    isUnhandled: Boolean,
+    sendThreads: ThreadSendPolicy,
+    projectPackages: Collection<String>,
+    logger: Logger,
     currentThread: java.lang.Thread = java.lang.Thread.currentThread(),
-    stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>> = java.lang.Thread.getAllStackTraces(),
-    logger: Logger = config.logger
+    stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>> = java.lang.Thread.getAllStackTraces()
 ) : JsonStream.Streamable {
 
-    internal val threads: MutableList<Thread>
+    internal constructor(
+        exc: Throwable?,
+        isUnhandled: Boolean,
+        config: ImmutableConfig
+    ) : this(exc, isUnhandled, config.sendThreads, config.projectPackages, config.logger)
```

Comment:
Do we need a secondary constructor?

I find this class very difficult to read - can we consolidate the constructors and make some of the parameters member variables?

Reply:
Discussed in-person and agreed to merge as-is since the other constructor is required for testing.